test/sequential/test-watch-mode.mjs

```diff
@@ -74,11 +80,31 @@ function assertRestartedCorrectly({ stdout, messages: { inner, completed, restar
   assert.deepStrictEqual(lines.slice(-end.length), end);
 }
 
+async function failWriteSucceed({ file, watchedFile }) {
+  let stdout = '';
+  const notFound = deferred();
```

Comment:
Not a fan of `deferred()` here - I think this can probably be accomplished more easily with stream operators.

Reply:
what stream do you refer to?

Reply:
Maybe something like this?

```js
// const rl = require('readline')
async function failWriteSucceed({ file, watchedFile }) {
  const child = spawn(execPath, ['--watch', '--no-warnings', file], { encoding: 'utf8' });
  try {
    // break the chunks into lines
    for await(const data of rl.createInterface({ input: child.stdout })) {
        if (data.startsWith('Completed running')) {
          break;
        }
        if (data.startsWith('Failed running')) {
          writeFileSync(watchedFile, 'console.log("test has ran");');
        }
    };
  } finally {
    child.kill();
  }
}
```